### PR TITLE
Tidied function address_in_close_group_range, and miraculously fixes …

### PR DESCRIPTION
**Pull request by s-brian**
…'not for us' output.

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/routing_node.rs [hunk 1]
```diff
@@ -1039,8 +1039,12 @@ impl<F> RoutingNode<F> where F: Interface {
             return true;
         }
 
-        let close_group = self.routing_table.our_close_group();
-        closer_to_target(&address, &self.routing_table.our_close_group().pop().unwrap().id(), &self.own_name)
+        match self.routing_table.our_close_group().pop() {
+            Some(furthest_close_node) => {
+                closer_to_target_or_equal(&address, &furthest_close_node.id(), &self.own_name)
```
Comment (dirvine) [inline src/routing_node.rs]: think the or_equal part may be where this is happening so multiple calls from same node is not closer but may be equal ? May be worth checking here with routing_table.check_node() call that we do not already have the node perhaps?

Comment (s-brian) [inline src/routing_node.rs]: Where what is happening? Turns out the' not for us' issue is a non issue just now anyway. The 'or_equal' is including the furthest close node as part of the group which both Ben and I think is correct.

Comment (dirvine) [inline src/routing_node.rs]: No worries, I was concerned we were seeing a node that matched here as OK to add to routing_table, if not that is totally correct. It was a quick scan so I am happy if you guys are.

Comment (s-brian) [inline src/routing_node.rs]: Got it, no, it occurs after that check. Thanks for clarifying.

Comment (dirvine) [inline src/routing_node.rs]: :+1:

Comment (benjaminbollen) [inline src/routing_node.rs]: Happy with it too.  Will Merge

**Pull request merged**